components/builder-depot/src/server.rs

```diff
@@ -2064,6 +2113,186 @@ pub fn download_latest_builder_key(req: &mut Request) -> IronResult<Response> {
     Ok(response)
 }
 
+pub fn create_origin_secret(req: &mut Request) -> IronResult<Response> {
+    let params = match validate_params(req, &["origin"]) {
+        Ok(p) => p,
+        Err(st) => return Ok(Response::with(st)),
+    };
+
+    let origin = &params["origin"];
+
+    let body = match req.get::<bodyparser::Struct<OriginSecretPayload>>() {
+        Ok(Some(body)) => {
+            if body.name.len() <= 0 {
+                return Ok(Response::with((
+                    status::UnprocessableEntity,
+                    "Missing value for field: `name`",
+                )));
+            }
+            if body.value.len() <= 0 {
+                return Ok(Response::with((
+                    status::UnprocessableEntity,
+                    "Missing value for field: `value`",
+                )));
+            }
+            body
+        }
+        _ => return Ok(Response::with(status::UnprocessableEntity)),
+    };
+
+    // get metadata from secret
+    let secret_metadata = match BoxKeyPair::secret_metadata(body.value.as_bytes()) {
+        Ok(res) => res,
+        Err(e) => {
+            return Ok(Response::with((
+                status::UnprocessableEntity,
+                format!("Failed to get metadata from payload: {}", e),
+            )))
+        }
+    };
+
+    debug!("Secret Metadata: {:?}", secret_metadata);
```

Comment:
Will this make private keys show up in our logs?

Reply:
No, it's just metadata about the encrypted payload. It will return the values here https://github.com/habitat-sh/core/blob/master/components/core/src/crypto/keys/box_key_pair.rs#L32-L37

Reply:
Ah, perfect.